extensions/amp-video/0.1/amp-video.js

```diff
@@ -397,7 +397,7 @@ export class AmpVideo extends AMP.BaseElement {
         this.video_.setAttribute('preload', 'auto');
       }
       pendingOriginPromise = this.getAmpDoc()
-        .whenFirstVisible()
+        .whenFirstPreviewedOrVisible()
```

Comment:
I believe this is a privacy leak. Shouldn't we only be connecting to an origin domain when we become visible?

Reply:
+1, `amp-video` will correctly play cached sources in `prerender` or `preview` mode, and then if needed play the origin sources when `visible`. I think this is the expected behavior, we shouldn't have to change anything to `amp-video`.